### DISCUSSION TITLE
Getting the gradients of joint targets using the XPBD integrator

### DISCUSSION THREAD

**Discussion by Manchewable** [NVIDIA/warp#146] [Q&A]
Hi,

I'm wondering whether it's possible to compute the gradients w.r.t. `model.joint_target` using the XPBD integrator.

Whenever I tried to get its gradients via `tape.backward()`, I'm getting a zero gradient in return. 

Appreciate to any insights!

**Comment by eric-heiden** [comment]
Hi @Manchewable,

We are currently in the process of improving the differentiability of the XPBD integrator. Differentiating w.r.t. joint targets will be possible in a future version. For now, you could try to ensure that you allocate a new state for every substep of the trajectory that is rolled out (so the states are not overwritten but instead `state_out` is mapped to a new state when you call `integrator.simulate()`), and that the Warp arrays in the state and model have `requires_grad=True` defined.
